Add feature for editors

Entenwilli opened the pull request:
This PR adds a feature for the required editors to edit PCM models.
Included are a feature importing the plugins required for the graphical and tree editors of the core PCM models, the datadictionary and the nodecharacteristics model.

@Nicolas-Boltz Due to additional dependencies, the build process (probably in the releng folder) needs to be adjusted as well.

Entenwilli committed:
Add feature for editors

Nicolas-Boltz committed:
add further dependencies

Entenwilli commented:
@Nicolas-Boltz: During my testing, I've discovered that the current URL for our analysis, specified in the targetplatform, points to `https://dataflowanalysis.github.io/updatesite/nightly/dataflowanalysis/`. This location however is invalid and the Maven build fails

Entenwilli commented:
The current issue is now fixed. The unit in the `features.xml` file was missing a `feaure` 😉

Entenwilli committed:
Add missing units to releng

Entenwilli commented:
The build should now be working.
So far, I've tested the following things in the created product:
- Running the tests of the dataflow analysis (without the analysis itself in the workspace)
- Opening PCM models
- Opening the Nodecharacteristics Tree Editor
- Editing the Data Dictionary

Is something else missing? If not, this PR can be reviewed, then merged! @Nicolas-Boltz

Nicolas-Boltz commented:
@Entenwilli Could you please copy the build workflow from the product branch. This way we can compare the product you have built with the one that is currently on the main branch.

Basically, I do not like the way the features are made up (as I have written in my last mail), where the plugins are defined in the editor and normal feature. Everything used by the product should only be defined as 'required'. 
Also we define features in the other projects/repos to make using the plugins easier (as the feature contains/defines all requirements). That is what I have done and focussed on in the other branch. 

Could you please check the same things with the product from the 'product'-branch? The current version is/will be deployed on the palladio updatesite ([updatesite.palladiosimulator.com/DataFlowAnalysis](https://updatesite.palladio-simulator.com/DataFlowAnalysis/product/branches/product/)).

Please also do what i described in https://github.com/DataFlowAnalysis/PCM-DataFlowAnalysis-Extension/issues/5 and change the product and targetplatform to only use the nodecharacteristic.feature

Nicolas-Boltz committed:
Update build.yml

Nicolas-Boltz committed:
fix missing .feature in naming

Entenwilli committed:
Remove editor feature

Entenwilli commented:
I've rebased the `product` branch onto this branch, removed the editor feature, and verified that the product is still working as intended.

Now, I will rename/create the nodecharacteristic feature that includes the editor, see issue https://github.com/DataFlowAnalysis/PCM-DataFlowAnalysis-Extension/issues/5

Entenwilli commented:
After creating the feature in https://github.com/DataFlowAnalysis/PCM-DataFlowAnalysis-Extension/issues/5, the updatesite (here on GitHub) does not contain the feature. 
@Nicolas-Boltz, I assume I've missed something during the creation of the feature. Would you take a look at the deployment of the updatesite?

Entenwilli committed:
Add feature for nodecharacteristics instead of plugins

Entenwilli committed:
Update feature file

Entenwilli committed:
Replace features from metamodel with unified feature

Entenwilli commented:
I've updated the target file for our build. However, now a different issue arrises:
During the creation of the product, the build fails with:
```
Installation failed.
An error occurred while installing the items
	session context was:(profile=DefaultProfile, phase=org.eclipse.equinox.internal.p2.engine.phases.Install, operand=null --> [R]com.ibm.icu 72.1.0.v20221115-2007, action=org.eclipse.equinox.internal.p2.touchpoint.eclipse.actions.InstallBundleAction).
	The artifact file for osgi.bundle,com.ibm.icu,72.1.0.v20221115-2007 was not found.
```
This seems to be an error during the tycho build of the product. @Nicolas-Boltz: Do you know what causes this issue?

Nicolas-Boltz commented on features/org.dataflowanalysis.product.feature/feature.xml:
```diff
@@ -163,6 +122,50 @@
+      <import feature="org.palladiosimulator.dataflow.confidentiality.pcm.feature" version="5.1.0" match="compatible"/>
    </requires>
 
+   <plugin
```
This Plugin definition needs to go

Nicolas-Boltz reviewed (changes requested):
See comment in feature.xml

Nicolas-Boltz commented:
@Entenwilli try making the feature change in the other "product" branch and look if there is an issue with the ibm package. 
Also I need to see the full error message to diagnose further. It could just be a updatesite of eclipse that is currently down (it happens sometimes...). You can just retry in a couple of hours.

Entenwilli committed:
Remove declared plugin in product feature

Entenwilli commented:
Removing the plugin from the product feature didn't help. Furthermore, i don't think it is caused by an updatesite being down.
Here is my [logfile](https://gist.github.com/Entenwilli/17614dcc70097b0ea6f474fb278a4b2d)

Nicolas-Boltz commented:
@Entenwilli The Plugin just needed to go in general, not to fix the error.

The build I started as an gh action seems to work fine, [see](https://github.com/DataFlowAnalysis/product/actions/runs/6691076379).
The problem seems to be on your side (could be the .m2 cache).
I case of strange errors you can always start the remote build to check.

Entenwilli commented:
Then, this PR should be ready to be merged. I removed the plugin either way
